### DISCUSSION TITLE
TSV file does not render, claiming "Illegal quoting", when it does not violate IANA standard

### DISCUSSION THREAD

**Discussion by kevinlul** [community/community#35389] [Other Feature Feedback, Questions, & Ideas]
GitHub can render TSVs as a table: https://docs.github.com/en/repositories/working-with-files/using-files/working-with-non-code-files#rendering-csv-and-tsv-data

However, GitHub rejects quotes in TSV fields, claiming "Illegal quoting", when they are not. Example case: https://github.com/DawnbrandBots/yaml-yugi-ko/blob/master/overrides.tsv

This problem was noted six years prior: isaacs/github#848
IANA `text/tab-separated-values` has no problem with quote characters at all! TSV is not CSV.

**Comment by glebm** [comment]
This is still an issue as of 7 Nov 2023.
IANA TSV only allows TAB as a delimiter, there are no escape characters.
Parsing it is as simple as `line.split("\t")`.

https://www.iana.org/assignments/media-types/text/tab-separated-values

**Reply by simon-clematide** [reply]
Would be very helpful for github-based data curation if this would finally work.

**Comment by zestyping** [comment]
TSV rendering in GitHub remains non-compliant as of September 2025.  GitHub produces the error message:

> We can make this file [beautiful and searchable](https://docs.github.com/articles/rendering-csv-and-tsv-data) if this error is corrected: Any value after quoted field isn't allowed in line 1.

To replicate this bug, try to render any .tsv file that contains a " (double-quote character) in the middle of a field.

  - https://github.com/zestyping/github-bugs/blob/main/minimal.tsv is a minimal example.
  - https://github.com/zestyping/github-bugs/blob/main/realistic.tsv is a realistic example (an excerpt from a real Anki card deck).

It appears that the GitHub renderer has expectations around the use of quotation marks, but that's incorrect; there is no such thing as a quoted field in a TSV file. The standard specifies that there are no quoting characters, and quotation marks have no special meaning:

https://www.iana.org/assignments/media-types/text/tab-separated-values

Please parse and render TSV files according to the IANA specification. Thanks!
